pkg/infrastructure/azure/storage.go

```diff
@@ -492,14 +505,51 @@ func CreateBlockBlob(ctx context.Context, in *CreateBlockBlobInput) (string, err
 		return "", fmt.Errorf("failed to create block blob: %w", err)
 	}
 
+	// SAS not supported when using managed identity.
+	if in.AuthType == azic.ManagedIdentityAuth {
+		return blockBlobClient.URL(), nil
+	}
```

Comment:
I don't get how this works. Is access to this url restricted in some way or is this a url open to the public internet? With SAS, IIUC the url includes the access key (so it needs to be protected), but I'm not sure what's going on here.

Reply:
The URL is restricted based on the storage account which has SAS disabled, so relies on Entra AD authentication.

Reply:
That sounds like a good solution, but we need changes to put that in place. 

Right now the storage containers are publicly open until #8944 merges. This PR depends on that publicly open storage account. When I test this, I can pull the ignition directly from my laptop with no authentication. When I test in combination with #8944, it breaks (because the container is now private and we're not providing access). 

The User Delegation SAS is a SAS that supports Entra ID, so that seems like the way forward.

Reply:
found an example here: https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob@v1.4.0/sas#example-package-UserDelegationSAS